### PR TITLE
Improvements for in-memory cache subsystem

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Backport of memory accounting improvements for in-memory (edge) cache from devel.
The PR makes the cache code identical to the one in devel, with the following exceptions:
* ideal lower and upper bound fill ratios for cache hash tables remain at 0.04 and 0.25. In devel they have been adjusted to 0.08 and 0.33.
* "introducedIn" version numbers for new startup options and metrics were changed from 3.12.0 to 3.11.3.
* in devel, the cache manager uses a "high water multiplier" value of 95% of the memory limit that is assigned to the cache. Previously, the cache manager effectively used a high water value of 0.7 * 0.8 = 56% of the memory limit. This old limit is kept for 3.11.
* when there is memory pressure, the cache manager will still to try to free memory in all caches, but it will not attempt to shrink tables. In devel, the cache manager also attempts to shrink the first x% of tables.

- [ ] :hankey: Bugfix
- [ ] :pizza: New feature
- [ ] :fire: Performance improvement
- [ ] :hammer: Refactoring/simplification

### Checklist

- [ ] Tests
  - [ ] **Regression tests**
  - [ ] C++ **Unit tests**
  - [ ] **integration tests**
  - [ ] **resilience tests**
- [ ] :book: CHANGELOG entry made
- [ ] :books: documentation written (release notes, API changes, ...)
- [ ] Backports
  - [x] Backport for 3.11: this PR
  - [ ] Backport for 3.10: -
  - [ ] Backport for 3.9: -

#### Related Information

- [ ] Docs PR: 
- [ ] Enterprise PR:
- [ ] GitHub issue / Jira ticket:
- [ ] Design document:

### CODE REVIEW

**Review by nerpaula — APPROVED** [review]
Metrics and startup options look good to me!

File: Documentation/Metrics/rocksdb_cache_edge_compressed_inserts_total.yaml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+name: rocksdb_cache_edge_compressed_inserts_total
+introducedIn: "3.11.3"
```
Comment (nerpaula) [inline Documentation/Metrics/rocksdb_cache_edge_compressed_inserts_total.yaml:2]: introduced in 3.11.3 or 3.11.4?

Comment (jsteemann) [reply Documentation/Metrics/rocksdb_cache_edge_compressed_inserts_total.yaml:2]: You are right, meanwhile it is 3.11.4.

File: Documentation/Metrics/rocksdb_cache_edge_empty_inserts_total.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+name: rocksdb_cache_edge_empty_inserts_total
+introducedIn: "3.11.3"
```
Comment (jsteemann) [inline Documentation/Metrics/rocksdb_cache_edge_empty_inserts_total.yaml:2]:
```suggestion
introducedIn: "3.11.4"
```

File: Documentation/Metrics/rocksdb_cache_edge_inserts_total.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
+name: rocksdb_cache_edge_inserts_total
+introducedIn: "3.11.3"
```
Comment (jsteemann) [inline Documentation/Metrics/rocksdb_cache_edge_inserts_total.yaml:2]:
```suggestion
introducedIn: "3.11.4"
```

**Review by MBkkt — APPROVED** [review]
I reviewed PRs to devel. I also looked this and it looks pretty same, so LGTM

**Pull request merged**